zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -2927,4 +2990,54 @@ private void removeChildListener(String path, IZkChildListener listener) {
       listeners.remove(listener);
     }
   }
+
+  interface ManipulateListener<T> {
+    void run() throws KeeperException, InterruptedException;
+  }
```

Comment:
Do we still need this? Can we simply use Runnable?

Reply:
Thanks for the suggestion. I think Runnable.run don't have exception signature? We need to handle these exceptions explicitly.

Reply:
Ok, that's a valid point. 
We can still use Callable with null return, to remove this wrapper. It's up to you.
(If you decide to stay with this interface, let's remove the generic, it's not used anyway)

Reply:
Yea already updated. :D